.circleci/config.yml

```diff
@@ -646,6 +597,47 @@ jobs:
             fi
           when: always
 
+  # Publishes new version onto npm
```

Comment:
Moving this block down makes PR review less convenient, I am afraid.

Here's what has changed. The following if/else block was removed (leaving the contents of the if block in place):

```
if [ -z "$CIRCLE_PULL_REQUEST" ]; then
 ...
else	
  echo "Skipping deploy."	
fi
```

The "save-*-cache" steps have been removed. No need to slow down a deploy while waiting for a cache to upload. This is already a noop in most cases because the cache is already up to date from the preceding jobs.

Consolidating `buck fetch` steps into a single script, matching `test_android`.